test-harness/pom.xml

```diff
@@ -0,0 +1,185 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.jenkins-ci.main</groupId>
+    <artifactId>pom</artifactId>
+    <version>1.645-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>jenkins-test-harness</artifactId>
+
+  <name>Test harness for Jenkins and plugins</name>
+  <description>Harness used to run functional tests of Jenkins core and plugins.</description>
+
+  <dependencies>
+    <dependency>
+      <!--
+        put hudson.war in the classpath. we can't pull in the war artifact directly
+        because Maven excludes all wars from classpath automatically. so we need a jar artifact.
+      -->
+      <groupId>${project.groupId}</groupId>
+      <artifactId>jenkins-war</artifactId>
+      <version>1.580.1</version>
+      <classifier>war-for-test</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jenkins-ci.modules</groupId>
+          <artifactId>sshd</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.mortbay.jetty</groupId>
+      <artifactId>jetty</artifactId>
+      <version>6.1.26</version>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-library</artifactId>
+      <version>1.3</version>
+    </dependency>
+    <dependency>
+      <groupId>net.sourceforge.htmlunit</groupId>
+      <artifactId>htmlunit</artifactId>
+      <version>2.18</version>
+      <exclusions>
+        <exclusion>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.jvnet.hudson</groupId>
+      <artifactId>embedded-rhino-debugger</artifactId>
+      <version>1.2</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jvnet.hudson</groupId>
+          <artifactId>htmlunit-core-js</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.netbeans.modules</groupId>
+      <artifactId>org-netbeans-insane</artifactId>
+      <version>RELEASE72</version>
+    </dependency>
+    <dependency>
+      <groupId>com.github.stephenc.findbugs</groupId>
+      <artifactId>findbugs-annotations</artifactId>
+      <version>1.3.9-1</version>
+    </dependency>
+    <dependency> <!-- TODO can we switch this to use Aether directly? -->
+      <groupId>org.jenkins-ci.lib</groupId>
+      <artifactId>lib-jenkins-maven-embedder</artifactId>
```

Comment:
Probably more bloated. Could use Aether directly, to avoid pulling in Maven classes.

Or could try to avoid doing anything like this in the test VM at all—prepare any necessary files before `surefire:test` using some new mojo in `maven-hpi-plugin`, and have `JenkinsRule` simply expect to find them in a prearranged place inside `target/test-classes/`, like we already do with `the.hpl`. This would probably be easier to adapt to novel situations, such as @KostyaSha’s work on ITs.

Either style of approach could potentially allow us to get rid of the `org.jenkins-ci-main:jenkins-war:war-for-test` hack in `WarExploder`, which would be a huge boon since this massive artifact is an exact copy of `org.jenkins-ci-main:jenkins-war:war` yet must be deployed and downloaded separately for every core release.